src/python/controller/controller.py

```
@@ -2,7 +2,7 @@
 
 import os
 from abc import ABC, abstractmethod
-from typing import List, Callable, Set
+from typing import Dict, List, Callable, Set
```

Ruff (0.15.1)

[warning] Import from `collections.abc` instead: `Callable`

Import from `collections.abc`

(UP035)